### PR TITLE
Add restaurant apis

### PR DESCRIPTION
**Pull request by Ogutu-Brian**
**What does this PR do?**
* Sets up django project for restaurant management
* Adds authentication APIs
* Adds  Authentication APIs

**How can this be manually tested?**
* After cloning the repo
```
$ git checkout ft-add-restaurant-apis
$ pipenv shell
$ pipenv install

$ psql postgres
$ create database restaurant_db
$ create role restaurant_user with encrypted password password
$ grant all privileges on database restaurant_db to restaurant_user

```
* Create a .env file in the root folder and paste the details in `.example-env` file
* run:
```
$ python manage.py runserver
```
* Using postman, test the endpoints in the documentation below

**What are the relevant links?**
[Link to postman documentation](https://documenter.getpostman.com/view/9068409/T1LJkoqy?version=latest#2f6f97c5-7e03-4b3b-b10b-824157d7e8d8)

**Running unit tests**
```
$ python manage.py test
```

### CODE REVIEW

File: restaurant/views.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  serializer = RestaurantSerializer(data=data)
+  serializer.is_valid(raise_exception=True)
+  serializer.save()
+
+  return Response(data=serializer.data, status=status.HTTP_201_CREATED)
+
+
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])
+def get_restaurants(request):
```
Comment (chaodev15) [inline restaurant/views.py]: Could you please replace this with ModelViewSet?
Please try to do this as much as you can.,

Comment (Ogutu-Brian) [reply restaurant/views.py]: Thanks, let me do that

**Comment by chaodev15** [comment]
thank you.

File: restaurant/views.py (hunk 1 above)
Comment (Ogutu-Brian) [reply restaurant/views.py]: > Could you please replace this with ModelViewSet?
> Please try to do this as much as you can.,

I have implemented on this view

File: restaurant/views.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    return Response(data=serializer.data, status=status.HTTP_200_OK)
+  except (ObjectDoesNotExist, ValidationError):
+    return Response({
+        'error': 'invalid ticket id'
+    }, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['PUT'])
+@permission_classes([AllowAny])
+def buy_ticket(request, ticket_id):
```
Comment (chaodev15) [inline restaurant/views.py]: please try to insert this method into the viewset

Comment (Ogutu-Brian) [reply restaurant/views.py]: > please try to insert this method into the viewset

Hello @chaodev15,

I have implemented that, kindly check it out.

Thanks, learning a lot and embracing the beauty of class based views

**Pull request merged**